### ISSUE TITLE
Metricbeat SQL Helper

### ISSUE THREAD

**Issue by sayden** [elastic/beats#9416]
Some of our modules for SQL databases rely on _plain_ SQL queries to fetch metrics information. Golang has the `sql` driver which should help in creating a SQL helper that just accepts queries and parses information for metricbeat.

There's a work already done in MS SQL metricbeat module that uses a similar approach, the first idea is to abstract this code from MSSQL module.

We have seen that if this is develop, the `vendor` folder in MySQL module should be removed https://github.com/elastic/beats/tree/master/metricbeat/module/mysql/vendor

We still need to discuss if this helper is going to be released **only as helper** of other modules **or** with a **sql-module too** but in any case, if a module if going to rely 90% of its functionality on this helper, it needs to follow a common modula GA checklist.

# SQL Helper release checklist

This checklist is intended for Devs which create or update a module to make sure modules are consistent.

## Modules

For a metricset to go GA, the following criterias should be met:

* [ ] Supported versions are documented
* [ ] Supported operating systems are documented (if applicable)
* [ ] Integration tests exist
* [ ] System tests exist
* [ ] Automated checks that all fields are documented
* [ ] Documentation
* [ ] Fields follow [ECS](https://github.com/elastic/ecs) and [naming conventions](https://www.elastic.co/guide/en/beats/devguide/master/event-conventions.html)
* [ ] Dashboards exists (if applicable)
* [ ] Kibana Home Tutorial (if applicable)

## Metricbeat module 

* [ ] Example `data.json` exists and an automated way to generate it exists (`go test -data`)
* [ ] Test environment in Docker exist for integration tests

## For this particular Metricbeat Helper

* [ ] Remove vendor folder from MySQL module once it is migrated to use the helper

**Comment by sayden** [comment]
Here I'll leave some links to comments that will help in the development of this module / helper:

https://github.com/elastic/beats/pull/9826#discussion_r244334584

**Comment by jsoriano** [comment]
https://github.com/elastic/beats/pull/13257 has been merged with an initial version of a generic module for SQL and I have created https://github.com/elastic/beats/issues/15048 as follow up. I am wondering now about the duplication with this issue.

Should we keep both and maintain this one for a helper? (a helper could still be helpful on the sql modules we already have)

Or we only continue only with the module and I close https://github.com/elastic/beats/issues/15048 as duplicated of this issue (#9416) and get it updated.
